src/merkle/path.rs

```diff
@@ -41,8 +42,8 @@ impl MerklePath {
     /// Verifies the Merkle opening proof towards the provided root.
     ///
     /// Returns `true` if `node` exists at `index` in a Merkle tree with `root`.
-    pub fn verify(&self, index: u64, node: Word, root: &Word) -> bool {
-        root == &self.compute_root(index, node)
+    pub fn verify(&self, index: u64, node: Word, root: &Word) -> Result<bool, MerkleError> {
+        self.compute_root(index, node).map(|v| v == *root)
```

Comment:
Ditto; merkle path verification is infallible and should be as simple as possible. If the provided index is somehow invalid, then it would just fail and return `false`. It doesn't mean however that we should just make assumptions about the index. We should, instead, just compute the root and compare, leaving the index problem to the tree implementation.